src/main/java/org/datadog/jenkins/plugins/datadog/clients/ClientFactory.java

```diff
@@ -31,9 +31,18 @@ of this software and associated documentation files (the "Software"), to deal
 import org.datadog.jenkins.plugins.datadog.DatadogUtilities;
 
 public class ClientFactory {
+    private static DatadogClient testClient;
```

Comment:
I like this idea, and how it eliminates use for PowerMock or lots of stubbing. I think if we want to unify the testing approach, we should make a separate PR on master with this, that way we can use it on all PRs. What do you think?

Reply:
Yeah let's do that on another PR